### PR TITLE
feat: structured log in JSON format

### PR DESCRIPTION
**Pull request by kuisathaverat**
## What does this PR do?

It Modifies the log output to be pure JSON.

## Why is it important?

We need a pure JSON output to enable log correlation on the APM UI.

### CODE REVIEW

File: bin/boot [hunk 1]
```diff
@@ -1,4 +1,4 @@
 #!/usr/bin/env sh
 set -ex
 bin/setup
-bundle exec foreman start
+bundle exec foreman start 2>&1 | cut -d '|' -f 2
```
Comment (kuisathaverat) [inline bin/boot]: I did not find any way to modify/remove the log output of `bundle`, so I `cut` the output

File: config/environments/production.rb [hunk 2]
```diff
@@ -83,7 +83,7 @@
   if true # ENV["RAILS_LOG_TO_STDOUT"].present?
     logger           = ActiveSupport::Logger.new(STDOUT)
     logger.formatter = config.log_formatter
-    config.logger    = ActiveSupport::TaggedLogging.new(logger)
+    config.logger    = ActiveSupport::Logger.new(STDOUT)
```
Comment (kuisathaverat) [inline config/environments/production.rb]: stdout output is easy to process with `cut`

File: config/initializers/lograge.rb [hunk 3]
```diff
@@ -4,7 +4,9 @@
 
   config.lograge.custom_options = lambda do |event|
     ElasticAPM.log_ids do |transaction_id, span_id, trace_id|
-      { :'transaction.id' => transaction_id,
+      { :'pid' => Process.pid,
```
Comment (kuisathaverat) [inline config/initializers/lograge.rb]: I've added some fields that are in the previous log output

File: docker-compose-elastic-cloud.yml [hunk 4]
```diff
@@ -0,0 +1,110 @@
+version: "2.1"
```
Comment (kuisathaverat) [inline docker-compose-elastic-cloud.yml]: this file is not related to structured log, is a docker-compose file to use Elastic Cloud as stack

File: docker-compose.yml [hunk 5]
```diff
@@ -4,7 +4,7 @@ services:
     build: .
     image: opbeans/opbeans-ruby:latest
     ports:
-      - "127.0.0.1:${OPBEANS_SERVER_PORT:-8000}:5000"
+      - "127.0.0.1:${OPBEANS_SERVER_PORT:-8000}:3000"
```
Comment (kuisathaverat) [inline docker-compose.yml]: normalize port with other Opbeans

File: docker-compose.yml [hunk 6]
```diff
@@ -16,19 +16,21 @@ services:
       - ELASTIC_APM_SERVER_URL=${ELASTIC_APM_SERVER_URL:-http://apm-server:8200}
       - ELASTIC_APM_JS_SERVER_URL=${ELASTIC_APM_JS_SERVER_URL:-http://localhost:8200}
       - ELASTIC_APM_JS_SERVICE_NAME=${ELASTIC_APM_JS_SERVICE_NAME:-opbeans-react}
-      - RAILS_ENV=test
-      - PGHOST=postgres
-      - PGDATABASE=opbeans
-      - PGUSER=postgres
-      - PGPASSWORD=hunter2
-      - PGPORT=5432
+      - RAILS_ENV=production
```
Comment (kuisathaverat) [inline docker-compose.yml]: enable production settings that contain structured log settings

File: docker-compose.yml [hunk 7]
```diff
@@ -133,6 +135,20 @@ services:
       retries: 10
       interval: 10s
 
+  redis:
```
Comment (kuisathaverat) [inline docker-compose.yml]: enable redis support

File: bin/boot [hunk 8]
```diff
@@ -1,4 +1,4 @@
 #!/usr/bin/env sh
 set -ex
```
Comment (v1v) [inline bin/boot]: When using pipes then

```
set -exo pipefail
```

File: docker-compose.yml (hunk 5 above)
Comment (mdelapenya) [reply docker-compose.yml]: Where are we setting the port at the ruby app? Because this is changing it at compose level, only

File: docker-compose-elastic-cloud.yml (hunk 4 above)
Comment (v1v) [reply docker-compose-elastic-cloud.yml]: For the record, 2.x is the version that enables the `depends_on:` directive with the `condition`

File: docker-compose.yml (hunk 5 above)
Comment (v1v) [reply docker-compose.yml]: ❤️ . It's now aligned with the ITs:
- https://github.com/elastic/apm-integration-testing/blob/1f929e1f772e6513c919165ea0a8ce636b876ac8/scripts/modules/opbeans.py#L578

Comment (mdelapenya) [reply docker-compose.yml]: Here it's where they port is configured: https://github.com/elastic/opbeans-ruby/search?q=5000&unscoped_q=5000

Comment (mdelapenya) [reply docker-compose.yml]: But Puma is using the right port: https://github.com/elastic/opbeans-ruby/blob/6422d8642436a0992d9d1638414ddb6f9f8b8e89/config/puma.rb#L12

File: tests/tests.bats [hunk 9]
```diff
@@ -29,11 +29,6 @@ CONTAINER="opbeans-ruby"
 	assert_output --partial 'healthy 0'
 }
 
-@test "test container is listening" {
```
Comment (kuisathaverat) [inline tests/tests.bats]: the console output is slower now so when the service is healthy the output is not there so the test fail, because the next test check the connectivity I have removed this one.
